src/utils/helper_embedding.py

```diff
@@ -16,3 +18,84 @@ def reshape_df(df: pd.DataFrame, col_name: str) -> pd.DataFrame:
         return df[['word', col_name]]
     except Exception as error:
         print(f"Cause: {error}")
+
+
+def calculate_cosine_similarity(a: list, b: list) -> float:
+    """
+    Computes the cosine-similarity of a word with each word in a dataframe.
+
+    :param a: List of first vector to compute cosine-similarity.
+    :param b: List of second vector to compute cosine-similarity.
+    :return: Float of the cosine-similarity between a and b.
+    """
+    try:
```

Comment:
I'd imagine (really just a guess) that pandas or sklearn would have an implementation of cosine similarity. It's probably a good learning exercise to implement it yourself but if there is an implementation in a much used package I'd go for that above writing one as it'll likely be optimised for speed and well tested so you wouldn't have to write your own

Reply:
Yup, I did look at sklearn and scipy but they seemed to be too complex for my usages! Maybe it was something i just needed to spend more time on. Possibly something I'll revisit in the future.